src/vc_util/src/lib.rs

```diff
@@ -553,16 +590,40 @@ mod tests {
             ID_ALIAS_CREDENTIAL_JWS,
             &test_canister_sig_pk().canister_id,
             &test_ic_root_pk_raw(),
+            CURRENT_TIME_BEFORE_EXPIRY_NS,
         )
         .expect("JWS verification failed");
     }
 
+    #[test]
+    fn should_fail_verify_credential_jws_if_expired() {
+        let result = verify_credential_jws_with_canister_id(
+            ID_ALIAS_CREDENTIAL_JWS,
+            &test_canister_sig_pk().canister_id,
+            &test_ic_root_pk_raw(),
+            CURRENT_TIME_AFTER_EXPIRY_NS,
+        );
+        assert_matches!(result, Err(e) if e.to_string().contains("credential expired"));
+    }
+
+    #[test]
+    fn should_fail_verify_credential_jws_if_no_expiry() {
+        let result = verify_credential_jws_with_canister_id(
+            TEST_CREDENTIAL_JWS_NO_EXPIRY,
+            &test_issuer_canister_sig_pk().canister_id,
+            &test_ic_root_pk_raw(),
+            CURRENT_TIME_AFTER_EXPIRY_NS,
+        );
+        assert_matches!(result, Err(e) if e.to_string().contains("structure is not semantically correct"));
```

Comment:
This error message is surprising to me: `"credential expired: the credential's structure is not semantically correct"`

As a dev, I would much rather read `"the credential's structure is not semantically correct: missing exp claim"` or something similar.

Reply:
Ah, ok.